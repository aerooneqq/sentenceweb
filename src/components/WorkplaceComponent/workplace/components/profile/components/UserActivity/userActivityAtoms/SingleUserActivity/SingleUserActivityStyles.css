.singleUserActivityCont{ 
    width: 16px;
    height: 16px;

    background-color: transparent;

    border-style: solid;
    border-radius: 5px;
    border-color: #e0e0e0;
    border-width: 1px;

    display: flex;
    justify-content: center; 
    align-items: center;

    position: relative;
}

    .singleUserActivityCont:hover{
        background-color: #e0e0e0;  
    }

    .singleUserActivityCont:hover .userActivityToolTip{
        visibility:  visible;  
    }

.activityBall{
    background-color: #F97646;
}

.userActivityToolTip{ 
    visibility: hidden;
    width: 30px;
    height: 20x;

    background-color: white;
    color: gray;
    text-align: center;
    font-size: 15px;

    border-style: solid;
    border-radius: 6px;
    border-width: 1px;
    border-color: #F97646;
    z-index: 2000;

    padding: 5px 0;

    position: absolute;
    bottom: 150%;
    left: 50%;
    margin-left: -15px;
}

    .userActivityToolTip::after{ 
        content: " ";
        position: absolute;
        top: 100%;
        left: 50%;
        margin-left: -5px;
        border-width: 5px;
        border-style: solid;
        border-color: gray transparent transparent transparent;
    }