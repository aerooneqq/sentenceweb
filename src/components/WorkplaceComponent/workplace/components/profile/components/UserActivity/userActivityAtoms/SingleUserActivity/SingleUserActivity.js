import React, {Component} from "react"

import "./SingleUserActivityStyles.css"

/**
 * PROPS LIST:
 * 1) activityData - the object which describes the activity
 * 2) handleSingleUserActivityClick
 */
export default class SingleUserActivity extends Component{ 
    constructor(props) { 
        super(props)

        this.state = { 
            count: props.activityData.count
        }

        this.maxCount = 15;

        this.getCountStyle = this.getCountStyle.bind(this)
        this.handleSingleActivityClick = this.handleSingleActivityClick.bind(this);
    }

    handleSingleActivityClick() { 
        this.props.handleSingleUserActivityClick(this.props.activityData);
    }

    render() { 
        return (
            <div className = "singleUserActivityCont" onClick = {this.handleSingleActivityClick}>
                <div style = {this.getCountStyle()} className = "activityBall" />
                <span class = "userActivityToolTip">
                    {this.state.count}
                </span> 
            </div>
        )
    }

    getCountStyle() {
        let size = this.state.count > this.maxCount ? this.maxCount : this.state.count;
        
        return { 
            "width": size + 1,
            "height": size + 1,
            "border-radius": size
        }
    }
}