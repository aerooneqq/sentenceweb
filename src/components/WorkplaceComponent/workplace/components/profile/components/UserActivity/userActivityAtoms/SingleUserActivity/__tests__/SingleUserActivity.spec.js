import React from "react";
import {shallow, render} from "enzyme"; 
import {shallowToJson} from "enzyme-to-json";
import SingleUserActivity from "../SingleUserActivity";

function handleSingleUserActivityClickDemo(x) { }

const props = { 
    activityData: { 
        count: 10
    },
    handleSingleUserActivityClick: handleSingleUserActivityClickDemo
}

describe("SingleUserActivity", () => { 
    it("should render corectly", () => { 
        expect(shallowToJson(shallow(<SingleUserActivity {...props}/>))).toMatchSnapshot();
    });

    it("should display right count", () => { 
        const singleUserActivity = shallow(<SingleUserActivity {...props}/>);

        expect(singleUserActivity.find("span").text()).toBe(props.activityData.count.toString());
    });

    it("should call the handleSingleActivityClick method when outter div os clicked", () => {
        const spy = jest.spyOn(SingleUserActivity.prototype, 'handleSingleActivityClick').mockImplementation(() => {});
        const singleUserActivity = shallow(<SingleUserActivity {...props} />);
        singleUserActivity.find("div.singleUserActivityCont").simulate("click");

        expect(spy).toBeCalled();
    });

    it("should call the handleSingleUserActivityClick method from props" + 
       "when handleSingleActivityClick is called", () => { 
        
        const spy = jest.spyOn(props, 'handleSingleUserActivityClick');
        const singleUserActivity = shallow(<SingleUserActivity {...props}/>);

        singleUserActivity.instance().handleSingleActivityClick();

        expect(spy).toBeCalled();
    });
});