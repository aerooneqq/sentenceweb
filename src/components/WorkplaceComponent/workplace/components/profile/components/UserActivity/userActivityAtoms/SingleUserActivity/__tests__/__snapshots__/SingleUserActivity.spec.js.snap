// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleUserActivity should render corectly 1`] = `
<div
  className="singleUserActivityCont"
  onClick={[Function]}
>
  <div
    className="activityBall"
    style={
      Object {
        "border-radius": 10,
        "height": 11,
        "width": 11,
      }
    }
  />
  <span
    class="userActivityToolTip"
  >
    10
  </span>
</div>
`;
